app/src/main/java/io/apicurio/registry/rest/RulesResourceImpl.java

```diff
@@ -113,7 +149,15 @@ public Rule updateGlobalRuleConfig(RuleType rule, Rule data) {
      */
     @Override
     public void deleteGlobalRule(RuleType rule) {
-        storage.deleteGlobalRule(rule);
+        try {
+            storage.deleteGlobalRule(rule);
+        } catch (RuleNotFoundException ruleNotFoundException) {
+            // This global rule doesn't exist in storage - if the rule also does not exist in
+            // the default global rules, throw the RuleNotFoundException
+            if (rulesProperties.getDefaultGlobalRuleConfiguration(rule) == null) {
+                throw ruleNotFoundException;
+            }
```

Comment:
But if the global rule **does** exist then the API responds with a 20x (success) response without having done anything.  Do we need a new response for that?  Some sort of HTTP conflict response?

Reply:
Good point - I have updated this to throw a new exception type (`DefaultRuleDeletionException`) and return a 409 Conflict response.